spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/config/KafkaBinderConfiguration.java

```diff
@@ -112,19 +114,14 @@ public AdminUtilsOperation kafka10AdminUtilsOperation() {
 		return new Kafka10AdminUtilsOperation();
 	}
 
-	private static Method getMethod(ClassLoader classLoader, String methodName) {
-		try {
-			Class<?> adminUtilClass = classLoader.loadClass("kafka.admin.AdminUtils");
-			Method[] declaredMethods = adminUtilClass.getDeclaredMethods();
-			for (Method m : declaredMethods) {
-				if (m.getName().equals(methodName)) {
-					return m;
-				}
+	private static Method getMethod(ClassLoader classLoader, String methodName) throws ClassNotFoundException {
+		Class<?> adminUtilClass = classLoader.loadClass("kafka.admin.AdminUtils");
```

Comment:
We should use `org.springframework.util.ClassUtils#forName` with a `null` classloader so that we observe the thread context classloader if set (among other things).